relay/sources/relayd/src/api.rs

```diff
@@ -28,26 +28,37 @@
 // You should have received a copy of the GNU General Public License
 // along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
 
+extern crate data_encoding;
+extern crate ring;
 use crate::error::Error;
 use crate::{configuration::LogComponent, stats::Stats, status::Status, JobConfig};
 use futures::Future;
 use slog::slog_info;
 use slog_scope::info;
 use std::collections::HashMap;
+use std::{fs, io};
 use std::{
     net::SocketAddr,
     sync::{Arc, RwLock},
 };
-use warp::{Filter};
+use warp::Filter;
 
-use crate::remote_run::{nodes_handle};
+use crate::remote_run::{nodes_handle, nodes_handle2, AgentParameters, RemoteRun, RemoteRunTarget};
+use crate::shared_files::{
+    compute_hash, metadata_writer, parse_hash_from_raw, parse_path_from_peek,metadata_hash_checker
+};
+use ring::digest::{Context, Digest, SHA1, SHA256, SHA512};
```

Comment:
I don't think we should use ring at all (it relies on C code that would make it harder to build and package)